### PR TITLE
[DOC-2229] feat(gsql): Doc for RBAC: Fine-grained query privilege: User Privilege Management;

### PR DESCRIPTION
**Pull request by jue-yuan**
[DOC-2229](https://graphsql.atlassian.net/browse/DOC-2229)
Doc to allow grant and revoke privileges on users directly.

[DOC-2229]: https://graphsql.atlassian.net/browse/DOC-2229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/user-access/pages/user-management.adoc [hunk 1]
```diff
@@ -268,6 +264,162 @@ GSQL > REVOKE ROLE role1, role2 ON GRAPH example_graph
 
 The above command will revoke roles `role1` and `role2` on graph `example_graph` from users `user1` and `user2`.
 
+== Grant privileges to a user
+
+Users with the `WRITE_USER` privileges on global can grant RBAC privileges to the users.
```
Comment (arun-TG-PM) [inline modules/user-access/pages/user-management.adoc]: Need to clarify better: Only users with the WRITE_USER privilege at the global level can grant RBAC privileges to other users.

File: modules/user-access/pages/user-management.adoc [hunk 2]
```diff
@@ -268,6 +264,162 @@ GSQL > REVOKE ROLE role1, role2 ON GRAPH example_graph
...
+
+. To grant privileges to a user, run the `GRANT PRIVILEGE` command from the GSQL shell:
++
+[source,text]
+----
+GSQL > GRANT WRITE ON ALL ROLES
+        IN GRAPH example_graph TO user1 , user2
+----
+
+This will allow `user1` and `user2` to modify roles on the graph `example_graph`. To see a full list of privileges and the command they allow users to run, see
```
Comment (arun-TG-PM) [inline modules/user-access/pages/user-management.adoc]: Be specific after this. It is not clear.

It should be like: This command allows user1 and user2 to modify roles on the example_graph. Specifically, they can perform operations that involve creating, modifying, or deleting roles within the graph.

File: modules/user-access/pages/user-management.adoc [hunk 3]
```diff
@@ -268,6 +264,162 @@ GSQL > REVOKE ROLE role1, role2 ON GRAPH example_graph
...
+GSQL > REVOKE WRITE ON ALL ROLES
+        IN GRAPH example_graph TO user1
+----
+
+This will revoke the `WRITE_ROLE` privilege from the user `user1` on graph `example_graph.`
+
+== Revoke type-level privileges
+You can revoke certain privileges from the type level with the `REVOKE PRIVILEGE` command.
+
+If the privilege has already been granted on the global scope or on the graph level, then revoking the privilege on the type level does not revoke the privilege in effect.
```
Comment (arun-TG-PM) [inline modules/user-access/pages/user-management.adoc]: This requires more clarity with an example.

Revoking a privilege at the type or attribute level does not affect privileges granted at higher levels (e.g., global or graph level). For instance, if a user has READ_DATA on a vertex type globally, revoking this privilege at the graph level will not prevent the user from accessing the vertex type elsewhere.

**Review by arun-TG-PM — COMMENTED** [review]
@jue-yuan - For each section that involves granting or revoking privileges, clearly explain what specific privileges are required to perform the operation. This will help users understand the prerequisites before attempting to grant or revoke privileges.

File: modules/user-access/pages/user-management.adoc (hunk 2 above)
Comment (jue-yuan) [reply modules/user-access/pages/user-management.adoc]: This is from our current docs in `role-management.adoc`. Whatever, I can improve it here.

File: modules/user-access/pages/user-management.adoc (hunk 3 above)
Comment (jue-yuan) [reply modules/user-access/pages/user-management.adoc]: Ditto.

**Comment by jue-yuan** [comment]
> @jue-yuan - For each section that involves granting or revoking privileges, clearly explain what specific privileges are required to perform the operation. This will help users understand the prerequisites before attempting to grant or revoke privileges.

The required privileges are already in the section `Require privilege`. Please refer to them.

File: modules/user-access/pages/user-management.adoc (hunk 1 above)
Comment (jue-yuan) [reply modules/user-access/pages/user-management.adoc]: Ditto.

Comment (victorleeTG) [inline modules/user-access/pages/user-management.adoc]:
```suggestion
Only users with the WRITE_USER privilege at the global level can grant RBAC privileges to other users.
```

File: modules/user-access/pages/user-management.adoc [hunk 4]
```diff
@@ -268,6 +264,162 @@ GSQL > REVOKE ROLE role1, role2 ON GRAPH example_graph
...
+
+=== Syntax
+
+[source,text]
+----
+GRANT <privilegeName1> (, <privilegeName2>)* ON <privilegeObjects>
+        [IN <privilegeScopes>]? TO <user_name1> (, <user_name2>)*
+----
+
+=== Require privilege
```
Comment (victorleeTG) [inline modules/user-access/pages/user-management.adoc]:
```suggestion
=== Required privilege
```

File: modules/user-access/pages/user-management.adoc (hunk 2 above)
Comment (victorleeTG) [inline modules/user-access/pages/user-management.adoc]:
```suggestion
This command will allow `user1` and `user2` to modify roles on the graph `example_graph`.
Specifically, they can create, modify, or delete roles for that graph.
To see a full list of privileges and the command they allow users to run, see 
```

File: modules/user-access/pages/user-management.adoc (hunk 3 above)
Comment (victorleeTG) [reply modules/user-access/pages/user-management.adoc]: @arun-TG-PM , if you are going to suggest exact wording, you might as well put it in as a suggested edit that can be accepted. Do you know how to do that?  Click the [+/-] icon, then revise the text between the quotes.

**Comment by victorleeTG** [comment]
There are still open suggestions for improvement. Those improvements should still be made, even though this PR is being accepted.

**Review by victorleeTG — APPROVED** [review]

**Pull request merged**